seasonal/README.md

```diff
@@ -0,0 +1,64 @@
+### Seasonal branding
+
+Each folder in this directory contains branding inspired by a specific season. Such assets are temporarily applied to the server to celebrate holidays, promote on-going events, or to show support for various world-wide movements.
+
+While the primary function is to organize the various assets we have, they are also used by **Seasonalbot's branding manager**. The bot will automatically apply seasonal assets pulled directly from this repository to the server. In order for Seasonalbot to properly detect and apply assets, certain conventions must be followed.
+
+#### Automatically managed assets
+
+There are 3 types of seasonal assets: **server icons**, **server banners**, and **bot avatars**. While a season always has **at most 1** banner and bot avatar, there may be **multiple** server icons. In such a case, Seasonalbot will periodically cycle available icons at a configurable interval (once every *n* days).
+
+In order for banners and avatars to be discovered, they shall be placed directly in a seasonal directory, named `seasonal/<season_name>/banner.png` and `.../avatar.png` respectively. Note that these assets are expected to *always* carry the **png** extension. Server icons shall always (regardless of whether the is only 1, or many) be placed in a nested directory, as follows: `.../server_icons/festive_256.gif`. Server icons are name and extension agnostic - they are registered simply by being present in the `server_icons/` directory.
```

Comment:
Is there any particular reason we're requiring banners and avatars to be png? Doesn't discord technically support other formats?

Reply:
It ensures consistent lossless image quality across the board, and is supported by pretty much every image editor/viewer out there. Encoding it with any other format is prone to differences in quality (compression levels, artifacting, etc). Not sure if this is _the_ reason, but it's _a_ reason.

Reply:
Most of the assets are vectorized, using a format with compression would be horrendous if we need to edit them later.

Reply:
Vectors still need to be converted to bitmaps if we want to use them anywhere.

Reply:
> @lemonsaurus: Is there any particular reason we're requiring banners and avatars to be png? Doesn't discord technically support other formats?

Technically, we could support them. However, I believe neither bot avatars nor banners can be animated at the moment. We do not currently have any static non-png assets, and I do not foresee us having them. The png format just appears to be the most fitting for this use case.

Therefore, I decided to make my life easier on Seasonalbot's side, and simply search for the full filename, including the extension.

If you believe that support for non-png assets should be added, then it only requires adjustments in the Seasonalbot repository.

Reply:
> This allowed me to avoid dealing with ambiguities such as banner.png and banner.jpg both being present.

Good point, I don't really think it's worth adding the support at that cost, so I'll resolve this.